### PR TITLE
[FrameworkBundle] Make StopWorkerOnSignalsListener configurable via messenger's config

### PR DESCRIPTION
**Pull request by rmikalkenas**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | Will prepare if PR gets merged

At the moment `Symfony\Component\Messenger\EventListener\StopWorkerOnSignalsListener` has hardcoded supported signals: SIGTERM, SIGINT. It's possible to pass custom ones, but it requires a compiler pass class which would override arguments. Adding a possibility to define it via messenger's config would make such configuration more easy.

cc @lyrixx

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -2028,6 +2028,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             $container->getDefinition('messenger.transport.beanstalkd.factory')->addTag('messenger.transport_factory');
         }
 
+        if (0 < \count($config['stop_worker_on_signals'])) {
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]:
```suggestion
        if ($config['stop_worker_on_signals']) {
```

**Review by lyrixx — COMMENTED** [review]
Thanks for your contribution!

It makes sens to me👍🏼

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 1 above)
Comment (rmikalkenas) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: updated

**Review by lyrixx — APPROVED** [review]
👍🏼 for the code. However, I'm not sure about the wording

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -1526,6 +1526,11 @@ function ($a) {
                                 ->thenInvalid('The "framework.messenger.reset_on_message" configuration option can be set to "true" only. To prevent services resetting after each message you can set the "--no-reset" option in "messenger:consume" command.')
                             ->end()
                         ->end()
+                        ->arrayNode('stop_worker_on_signals')
+                            ->defaultValue([])
+                            ->info('A list of signals to stop worker. If not set - use defaults defined in listener.')
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: we should be explicit here instead of referencing the listener

Comment (rmikalkenas) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: @nicolas-grekas updated with clarification of default signals

**Comment by nicolas-grekas** [comment]
Thank you @rmikalkenas.

**Pull request merged**